src/sentry/signals.py

```diff
@@ -21,14 +26,14 @@ class receivers_raise_on_send:
 
     receivers: Any
 
-    def __init__(self, receivers: Any | List[Any] = _all):
+    def __init__(self, receivers: _AllReceivers | Receiver | List[Receiver] = _AllReceivers.ALL):
         self.receivers = receivers
 
     def __enter__(self) -> None:
         global _receivers_that_raise
         self.old = _receivers_that_raise
 
-        if self.receivers is _all:
+        if isinstance(self.receivers, _AllReceivers):
```

Comment:
oh this can now become `self.receivers is _Receivers.All` -- mypy has exhaustive enum checks so it'll know that the other side of the union is taken

Reply:
here's an example https://github.com/pre-commit/pre-commit/blob/c25ea47cf78b42da8fd403c435f156b056aaeb38/pre_commit/envcontext.py#L52